File: 1st_week/10828.py
# Baekjoon Online Judge Q)10828
# Language : python3
# Version : 3.5.2
# Date : 2017. 03. 06
# Junyoung Jung(@sauber92)
import sys

loop = int(input())
stack = []
length = len(stack)

if loop < 0 or loop > 10000:
    sys.exit()

for i in range(loop):
    value = input().split()

    if value[0] == "push":
        stack.append(int(value[1]))
        length += 1

    elif value[0] == "pop":
        if length == 0:
            print(-1)
        else:
            print(stack[-1])
            stack.pop()
            length -= 1

    elif value[0] == "size":
        print(length)


    elif value[0] == "empty":
        if length == 0:
            print(1)
        else:
            print(0)

    elif value[0] == "top":
        if length == 0:
            print(-1)
        else:
            print(stack[-1])

File: Stack/10828.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {
    int n, x;
    string s;
    stack<int> st;

    cin >> n;

    for (int i = 0; i < n; i++) {
        cin >> s;

        if (s == "push") {
            cin >> x;
            st.push(x);
        }
        else if (s == "pop") {
            if (!st.empty()) {
                cout << st.top() << endl;
                st.pop();
            }
            else {
                cout << "-1" << endl;
            }
        }
        else if (s == "size") {
            cout << st.size() << endl;
        }
        else if (s == "empty") {
            if (!st.empty()) {
                cout << "0" << endl;
            }
            else {
                cout << "1" << endl;
            }
        }
        else if (s == "top") {
            if (!st.empty()) {
                cout << st.top() << endl;
            }
            else {
                cout << "-1" << endl;
            }
        }
    }

    return 0;
}
